auction-server/src/auction/service/verification.rs

```diff
@@ -673,11 +783,16 @@ impl Verification<Svm> for Service<Svm> {
             transaction:        bid.chain_data.transaction.clone(),
         };
         let permission_key = bid_chain_data.get_permission_key();
+        let submit_type = self.get_submission_state(&permission_key).await;
```

Comment:
Basically we want to make sure the swap_instruction only exists for the SubmitType::ByOther. Isn't it better to include this logic inside the `extract_bid_data`?